File: frontend/src/api/salary.ts
import { api } from "./client";

export interface SalaryRevision {
  id: string;
  employee_id: string;
  effective_date: string;
  previous_salary: string | null;
  revised_salary: string;
  revision_percentage: string | null;
  comments: string | null;
  approval_status: string;
}

export interface CurrentSalary {
  current_salary: string | null;
  latest_revision_date: string | null;
}

export const salaryApi = {
  history: (employeeId?: string) =>
    api.get<SalaryRevision[]>(
      `/salary/history${employeeId ? `?employee_id=${employeeId}` : ""}`,
    ),
  current: (employeeId?: string) =>
    api.get<CurrentSalary>(
      `/salary/current${employeeId ? `?employee_id=${employeeId}` : ""}`,
    ),
  addRevision: (data: {
    employee_id: string;
    effective_date: string;
    previous_salary?: string;
    revised_salary: string;
    revision_percentage?: string;
    comments?: string;
    approval_status?: string;
  }) => api.post<SalaryRevision>("/salary/revisions", data),
};

File: backend/app/schemas/salary.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.models.enums import ApprovalStatus


class SalaryRevisionCreate(BaseModel):
    employee_id: uuid.UUID
    effective_date: date
    previous_salary: Decimal | None = None
    revised_salary: Decimal
    revision_percentage: Decimal | None = None
    comments: str | None = None


class SalaryRevisionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    employee_id: uuid.UUID
    effective_date: date
    previous_salary: Decimal | None = None
    revised_salary: Decimal
    revision_percentage: Decimal | None = None
    comments: str | None = None
    approval_status: ApprovalStatus


class CurrentSalaryOut(BaseModel):
    """Derived current salary = latest approved revision."""

    current_salary: Decimal | None = None
    latest_revision_date: date | None = None
